ddsreplayer/test/blackbox/mcap/dds/HelloWorldSubscriber.h

```diff
@@ -0,0 +1,116 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file HelloWorldSubscriber.h
+ *
+ */
+
+#ifndef _EPROSIMA_DDSRECORDER_RESOURCES_DDS_TYPEINTROSPECTION_TYPEINTROSPECTIONSUBSCRIBER_H_
+#define _EPROSIMA_DDSRECORDER_RESOURCES_DDS_TYPEINTROSPECTION_TYPEINTROSPECTIONSUBSCRIBER_H_
+
+#include <atomic>
+#include <condition_variable>
+#include <mutex>
+
+#include <fastdds/dds/core/status/SubscriptionMatchedStatus.hpp>
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/domain/DomainParticipantListener.hpp>
+#include <fastdds/dds/subscriber/DataReaderListener.hpp>
+
+#include "types/hello_world/HelloWorldPubSubTypes.h"
+#include "data_to_check.idl"
+
+/**
+ * @brief Class used to group into a single working unit a Subscriber with a DataReader and its listener.
+ *
+ */
+class HelloWorldSubscriber : public eprosima::fastdds::dds::DomainParticipantListener
+{
+public:
+
+    /**
+     * @brief Construct a new Type Lookup Service Subscriber object
+     *
+     * @param topic_name Name of the DDS Topic
+     * @param domain DDS Domain of the DomainParticipant
+     * @param max_messages Number of messages to be received before triggering termination of execution
+     */
+    HelloWorldSubscriber(
+            const std::string& topic_name,
+            uint32_t domain,
+            uint32_t max_messages,
+            DataToCheck& data);
+
+    /**
+     * @brief Destroy the Type Lookup Service Publisher object
+     *
+     */
+    virtual ~HelloWorldSubscriber();
+
+    /**
+     * @brief Run the subscriber until "number" samples are received
+     *
+     */void run();
+
+    //! Set the maximum number of messages to receive before exiting
+    void set_max_messages(
+            uint32_t max_messages);
```

Comment:
I would simplify the subscriber, we won't be needing this feature.

Reply:
What I meant is not to use the mechanism of destroying the subscriber once max_messages are received, it increases complexity. In my opinion, it would be better to wait in a thread until the replayer finishes processing the mcap file and emits a notification.